libraries/anvill_passes/src/RecoverStackFrameInformation.cpp

```diff
@@ -274,18 +300,51 @@ RecoverStackFrameInformation::UpdateFunction(
   // Pre-initialize the stack frame to zero if we have been requested
   // to do so. From the whole FunctionPass class, we get the setting
   // from the LiftingOptions class
-  if (init_strategy == StackFrameStructureInitializationProcedure::kZeroes) {
-    auto null_value = llvm::Constant::getNullValue(stack_frame_type);
-    builder.CreateStore(null_value, stack_frame_alloca);
+  switch (init_strategy) {
+    case StackFrameStructureInitializationProcedure::kZeroes: {
+      auto null_value = llvm::Constant::getNullValue(stack_frame_type);
+      builder.CreateStore(null_value, stack_frame_alloca);
 
-  } else if (init_strategy ==
-             StackFrameStructureInitializationProcedure::kUndef) {
-    auto null_value = llvm::UndefValue::get(stack_frame_type);
-    builder.CreateStore(null_value, stack_frame_alloca);
+      break;
+    }
 
-  } else if (init_strategy !=
-             StackFrameStructureInitializationProcedure::kNone) {
-    return StackAnalysisErrorCode::InvalidParameter;
+    case StackFrameStructureInitializationProcedure::kUndef: {
+      auto null_value = llvm::UndefValue::get(stack_frame_type);
+      builder.CreateStore(null_value, stack_frame_alloca);
+
+      break;
+    }
+
+    case StackFrameStructureInitializationProcedure::kSymbolic: {
```

Comment:
Can you add a comment here, describing what this is doing and why.

Reply:
I've added a "look at the enum definition" comment here and documented the strategies in more details in the public interface